espei/paramselect.py

```diff
@@ -1077,113 +418,66 @@ def estimate_hyperplane(dbf, comps, phases, current_statevars, comp_dicts,
-                """
-                template_error = textwrap.dedent(template_error)
-                # print('Dumping', 'error-'+str(error_time)+'.py')
-                # with open('error-'+str(error_time)+'.py', 'w') as f:
-                #    f.write(template_error.format(dbf.to_string(fmt='tdb'), comps, phases, cond_dict, {key: float(x) for key, x in parameters.items()}))
-            # print('MULTI_EQDATA', multi_eqdata)
+            multi_eqdata = equilibrium(dbf, comps, phases, cond_dict, verbose=False,
+                                       model=phase_models, scheduler=dask.local.get_sync, parameters=parameters)
             # Does there exist only a single phase in the result with zero internal degrees of freedom?
             # We should exclude those chemical potentials from the average because they are meaningless.
             num_phases = len(np.squeeze(multi_eqdata['Phase'].values != ''))
-            zero_dof = np.all((multi_eqdata['Y'].values == 1.) | np.isnan(
-                multi_eqdata['Y'].values))
+            zero_dof = np.all((multi_eqdata['Y'].values == 1.) | np.isnan(multi_eqdata['Y'].values))
             if (num_phases == 1) and zero_dof:
-                region_chemical_potentials.append(
-                    np.full_like(np.squeeze(multi_eqdata['MU'].values), np.nan))
+                region_chemical_potentials.append(np.full_like(np.squeeze(multi_eqdata['MU'].values), np.nan))
             else:
-                region_chemical_potentials.append(
-                    np.squeeze(multi_eqdata['MU'].values))
-    # print('REGION_CHEMICAL_POTENTIALS', region_chemical_potentials)
-    region_chemical_potentials = np.nanmean(region_chemical_potentials, axis=0,
-                                            dtype=np.float)
+                region_chemical_potentials.append(np.squeeze(multi_eqdata['MU'].values))
+    region_chemical_potentials = np.nanmean(region_chemical_potentials, axis=0, dtype=np.float)
     return region_chemical_potentials
 
 
-def tieline_error(dbf, comps, current_phase, cond_dict,
-                  region_chemical_potentials, phase_flag,
-                  phase_models, parameters):
-    # print('COND_DICT ({})'.format(current_phase), cond_dict)
-    # print('PHASE FLAG', phase_flag)
+def tieline_error(dbf, comps, current_phase, cond_dict, region_chemical_potentials, phase_flag,
+                  phase_models, parameters, debug_mode=False):
     if np.any(np.isnan(list(cond_dict.values()))):
         # We don't actually know the phase composition here, so we estimate it
         single_eqdata = calculate(dbf, comps, [current_phase],
                                   T=cond_dict[v.T], P=cond_dict[v.P],
-                                  model=phase_models, parameters=parameters,
-                                  pdens=10)
-        # print('SINGLE_EQDATA (UNKNOWN COMP)', single_eqdata)
+                                  model=phase_models, parameters=parameters, pdens=10)
```

Comment:
This was my mistake: running with `pdens` this low may actually cause problems for multi-sublattice solidus composition estimates. It's computing driving force using an unrefined point sample and then taking the minimum. A really coarse sample may lead to poor estimates.